app/celery/process_delivery_status_result_tasks.py

```diff
@@ -1,13 +1,296 @@
-from flask import current_app
-from app import notify_celery
+import math
+import time
+import datetime
+from app.celery.service_callback_tasks import check_and_queue_callback_task
 from app.celery.process_pinpoint_inbound_sms import CeleryEvent
 
+from app.dao.notifications_dao import (
+    dao_get_notification_by_reference,
+    dao_update_notification,
+    update_notification_status_by_id,
+)
+
+from typing import Tuple
+from celery.exceptions import Retry
+from flask import current_app
+
+from notifications_utils.statsd_decorators import statsd
+from sqlalchemy.orm.exc import NoResultFound, MultipleResultsFound
+from app import notify_celery, statsd_client, clients, DATETIME_FORMAT
+from app.config import QueueNames
+from app.dao.service_callback_dao import dao_get_callback_include_payload_status
+
+from app.models import (
+    NOTIFICATION_DELIVERED,
+    NOTIFICATION_TECHNICAL_FAILURE,
+    NOTIFICATION_PERMANENT_FAILURE,
+    Notification,
+    NOTIFICATION_PREFERENCES_DECLINED,
+)
+
+FINAL_STATUS_STATES = [
+    NOTIFICATION_DELIVERED,
+    NOTIFICATION_PERMANENT_FAILURE,
+    NOTIFICATION_TECHNICAL_FAILURE,
+    NOTIFICATION_PREFERENCES_DECLINED,
+]
+
 
 # Create SQS Queue for Process Deliver Status.
-@notify_celery.task(bind=True, name="process-delivery-status-result", max_retries=48, default_retry_delay=300)
+@notify_celery.task(bind=True, name="process-delivery-status-result", max_retries=48, default_retry_delay=300, )
+@statsd(namespace="tasks")
 def process_delivery_status(self, event: CeleryEvent) -> bool:
-    current_app.logger.info('processing delivery status: %s', event)
-    from app import clients
-    provider = clients.get_sms_client('twilio')
-    current_app.logger.info('Translated message: %s', provider.translate_delivery_status(event['message']))
+
+    """Celery task for updating the delivery status of a notification"""
+
+    # preset variables to address "unbounded local variable"
+    sqs_message = None
+    notification_platform_status = None
+
+    # log that we are processing the delivery status
+    current_app.logger.info("processing delivery status: %s", event)
```

Comment:
This should be broken into two lines. We don't want the `event` in the logs unless we're in a lower environment or we've hit a critical error. So one line to log that we're in this method is great, then add a second line to call `logger.debug` with the event.